### DISCUSSION TITLE
Fix New Tumblr Layout

### DISCUSSION THREAD

**Discussion by TDLBallistic** [AprilSylph/XKit-Rewritten#1150] [Suggestion Box]
![7-19-2023 5-35-10 AM](https://github.com/AprilSylph/XKit-Rewritten/assets/60634610/e59fd513-f356-4ac7-9553-e95b4a6f2133)
So tumblr just decided to wholesale copy Twitter's homework. They say it's experimental, but they've said that about a lot of things they've kept and made permanent against literally everyone's wishes.

Please just put my right side sidebar back I'm begging you

**Comment by arcanetrivia** [comment]
I feel like the right-hand sidebar's been like that for a while, but the things being on the left instead of on the top is kind of strange. OTOH one of my peeves with Tumblr has long been wasted space in a widescreen viewport, so....

**Reply by TDLBallistic** [reply]
The recommended blogs and radar have, but all of our account information was over there before. Now they swapped sides and hid everything in mobile style collapsible text. I hate it.

**Comment by slach123** [comment]
This is super frustrating for me, I like to use the right hand side of my screen to watch a video, but if I pull the tumblr window to one side it does the condensed version where I don't get access to any menu at all. The off center dash was great for me. Hope we can restore with xkit soon 😭

**Reply by marcustyphoon** [reply]
> if I pull the tumblr window to one side it does the condensed version where I don't get access to any menu at all

This is true with the current layout with the header, right? In both cases you have to click an icon to access the menu.

**Reply by slach123** [reply]
Yes, but on the previous layout, the dash was off-centered, so there was enough space on the right-hand side to have a popped out video while in the new version the dash is directly in the middle of the page, minimizing the amount of space. Here are pictures demonstrating. The video window is the same size between the 2 images.
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/95504277/27704f91-12b6-46bb-ad34-b46c91a8d59f)
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/95504277/8cedfe66-ad43-405c-9106-1fe2e01df69d)

**Reply by marcustyphoon** [reply]
> if I pull the tumblr window to one side it does the condensed version

Are you referring to making the Tumblr window smaller, like this?

(Or is the video you're playing stuck inside the browser window, and if so, what are you using to make it do that?)

![](https://github.com/AprilSylph/XKit-Rewritten/assets/8336245/49e66c7b-c35c-4f55-821b-5e3a047589c5)

**Comment by kaymarierose** [comment]
It really is just a clutter of information and super unpleasant to look at. I very much prefer the simplistic small icons at the top and out of the way. This currently layout is too much information, too much text. It's overloading. Please, make it stop.

**Comment by rosymaplemelody** [comment]
Is it possible to make the sidebars adjustable? So we can maybe click and drag them to smaller margins on the side of the screen? Or create an option for a minimized left side-bar that's a series of icons down the side of the page, rather than a gigantic box of text?

**Reply by slach123** [reply]
I think this would be a good compromise. A minimized version would be nice, or even just sliding the left-hand menu farther to the left. The dash is currently centered, with the left- and right-hand sidebars sticking to the edges, with white (blue) space appearing at both outside borders evenly. Having an option to anchor to one side might be a solution. I personally prefer the tumblr dashboard to be off-center on my screen, so even if there was just an option to have the sidebar hug the left edge to keep the off centered dashboard, I would prefer that. Currently, condensing the window far enough does give "minimized" icons, so that should be doable. (Images attached to show the empty space shrinking evenly, and the look of the "minimized" icons.

![image](https://github.com/AprilSylph/XKit-Rewritten/assets/95504277/fafe66c7-8f0e-4dd2-bcff-2c19cef8061a)
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/95504277/8ffd48ef-a332-41f9-a20c-bbacc84a2db7)
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/95504277/fd3bc651-c0b4-47a1-8167-b94ae4d42002)
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/95504277/bb7375c2-99b6-48fd-80f1-36531e7381f8)

**Comment by IamJustice743** [comment]
I personally just don't want a left-hand sidebar at all, I won't lie. I much prefer the toolbar menu to be placed at the top of the screen with blog settings easily accessible through a menu drop-down. I don't know how possible it would be to bring back this setup, but if it is possible I would be greatly indebted to the coders that make it happen

**Comment by PBartolM** [comment]
If this is where we talk about that sidebar, would it be possible to make the "(pencil icon) create" button be a less distinct color? like the white/gray of the other options in the sidebar? I just dont like having something bright on the corner of my vision while browsing.

**Comment by krc** [comment]
Yeah I don't know how possible it would be to fully revert things, but can you try? I don't want the new layout improved, I want the old layout. The new one is horrible and is not tumblr it's **** twitter. And they are taking it out of test and making it the standard now.

**Comment by arcanetrivia** [comment]
Finally got this rolled out to me today. I don't think I hate it as much as a lot of people (kind of in that area of "don't like it but I'll probably manage"), but I still want the center column to be wider. (This offended me more actually when the buttons that are now on the left were on the top, because all that area was blank, a real waste of the viewport in a 1920x1080 maximized window.) Panorama used to go too far the other way, but now... it... does nothing at all? Last news appears to be "being worked on" (#845) but idk how that's coming along, and this change probably doesn't help.

**Reply by marcustyphoon** [reply]
I'll get to it, if it's reasonably easy. Haven't taken a look yet. Lots of other stuff to work on first though!

**Reply by arcanetrivia** [reply]
🥳 At long last! I guess it wasn't reasonably easy. 😉 

For a few months I'd been using Dashboard Plus's "content width" tweak for this, which provides similar base functionality, but behaves differently in a few ways. I like the way D+ allows me to specify any arbitrary pixel width, but I get the idea from the pull requests related to Panorama that a dropdown with certain predefined choices was for predictability in the results, which makes sense. I do like Panorama's feature to _not_ expand media to the full width of the container, as that tends to look bad. And bless whosever idea it was to be able to widen the columns in Patio, because Patio has basically become my Drafts replacement, but because the "thumbnails" layout mysteriously still isn't compatible with the Queue, it had a similar "wasted space" problem in 1920x1080.
![image](https://github.com/user-attachments/assets/dc3d4aaf-76a6-43e1-adf1-43a240536605)
![image](https://github.com/user-attachments/assets/66454f67-594a-46ae-bbb2-f9f884ace130)
(I mean there's still a lot of empty space there, but that's where the Queue would go if I could, dangit!)

**Reply by marcustyphoon** [reply]
Hah, yeah, it was pretty painful. There are still a few pages it randomly doesn't work on, too, which... once again, hopefully we'll get to those eventually, but don't hold your breath!

Glad you like it! I tried to make it as "clean" an adjustment as I could, so that it could be configured to feel like a Tumblr update rather than necessarily always looking like something that came from an extension.

**Reply by enchanted-sword** [reply]
This was an absolutely ancient thread to get an email for lol

**Comment by enchanted-sword** [comment]
This is certainly fixable (I've done it myself), but the question is whether or not XKit plans on setting a precedent of providing such a feature, and likely by extension implementing `obfuscatedFeatures` toggles (#1189).

**Comment by nightpool** [comment]
I think any feature we implement should be based on CSS / JS modification,
I would be hesitant to rely on the A/B test sticking around now that
they've announced publicly that they'll be moving forward with the new
sidebar

**Reply by enchanted-sword** [reply]
To quote my reply from #1189;
>...by default, every single flag is enabled. Rather than being an option to disable certain features when enabled, these flags for UI updates instead represent additions onto some "base" version of the UI. In the case of features currently undergoing A/B testing, these flags are simply not present in the obfuscatedFeatures property of blogs that are not part of the A/B test. This structuring implies there being a decent system in place for rolling back UI updates, as has occasionally happened in the past.

It's not so much an A/B test flag as it is just a general flag that was only applied to selective blogs before today. The UI updates that have toggleable flags extend back several years, at least to the point before the previous two post footer updates.
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/81981111/8944ca72-d614-4f22-ae2c-e02f0c8a114e)
To quote myself again;
> There are definitely advantages to using declarative code to revert UI changes. However, Tumblr's feature testing is rather unpredictable, and for me at least, simply disabling the update seemed like a more practical option than continuing to updating my script every time something gets moved in the UI. And just moving stuff around isn't enough once Tumblr starts changing event- and data-related elements; sure, I could probably re-add the button and event that changes the palette to the account submenu, but I wouldn't want to tackle the activity feed, messaging updates, or the disappearance of the legacy editor. A more experienced person probably could, but at a certain point, is it worth it to make so many changes to the document and document events just for the sake of the feature being toggleable with a single click?

Even now that the update is out of the testing phase I would be surprised if Tumblr didn't continue making regular tweaks to it that would make a pure CSS / JS modification difficult to keep up-to-date.

**Reply by marcustyphoon** [reply]
(I think we could—and shouldn't, any more than necessary—get into the semantics: the A/B tests are "`obfuscatedFeatures` contains this flag" vs "`obfuscatedFeatures` doesn't contain this flag." Your point that parts of the codebase often remain flagged, with older alternative pieces of code still shipped to the user on the falsy side of the flag, for extended periods of time is certainly valid.)

**Comment by TDLBallistic** [comment]
Ayo anyone still looking around here, someone did make a Style to have Tumblr look like it used to!
https://userstyles.world/style/11286/old-tumblr-dashboard-july-2023

**Reply by lilybuddy** [reply]
yooooo this is what i was hoping for, thank you SO much! that new dashboard is so distracting, it's not ADHD/autism friendly AT ALL.

**Comment by arcanetrivia** [comment]
There's also this https://github.com/enchanted-sword/dashboard-unfucker but one version seems to make use of the flag that opts out of features, which I guess will not work in the not too distant future?

**Reply by enchanted-sword** [reply]
Because of how Tumblr handles UI, updates I'm relatively confident that the flag method will still work for a while to come.

**Comment by enchanted-sword** [comment]
I'm going to start coding a fix based on my non-window property script. If someone's going to do it, I've already done it once so I might as well do it again.

**Reply by enchanted-sword** [reply]
Done, #1217

**Reply by krc** [reply]
hope this can get merged. I'd rather just have the one extension than extension plus userscript extension plus userscript (I don't need tampermonkey/etc for anything else)... in the mean time, glad Dashboard Unfucker is available! Thanks for all your work!
